src/main/java/io/fabric8/maven/docker/config/handler/compose/DockerComposeServiceWrapper.java

```diff
@@ -396,4 +406,97 @@ private Arguments asArguments(Object command, String label) {
     private void throwIllegalArgumentException(String msg) {
         throw new IllegalArgumentException(String.format("%s: %s - ", composeFile, name) + msg);
     }
+
+    /**
+     * Examines the supplied {@code volume} string, and resolves relative paths.  The returned string is guaranteed to
+     * be absolute, but <em>not</em> guaranteed to exist.
+     * <p>
+     * Volumes may be defined inside of {@code service} blocks <a href="https://docs.docker.com/compose/compose-file/compose-file-v2/#volumes-volume_driver">
+     * as documented here</a>:
+     * </p>
+     * <pre>
+     * volumes:
+     * # Just specify a path and let the Engine create a volume
+     * - /var/lib/mysql
+     *
+     * # Specify an absolute path mapping
+     * - /opt/data:/var/lib/mysql
+     *
+     * # Path on the host, relative to the Compose file
+     * - ./cache:/tmp/cache
+     *
+     * # User-relative path
+     * - ~/configs:/etc/configs/:ro
+     *
+     * # Named volume
+     * - datavolume:/var/lib/mysql"
+     * </pre>
+     * <p>
+     * This method only operates on volume strings that are relative: beginning with {@code ./} or {@code ~/}.  Relative
+     * paths beginning with {@code ./} are absolutized relative to the supplied base directory, which <em>must</em> be
+     * absolute.  Paths beginning with {@code ~/} are interpreted relative to {@code new File(System.getProperty(
+     * "user.home"))}, and {@code baseDir} is ignored.
+     * </p>
+     * <p>
+     * Volume strings that do not begin with a {@code ./} or {@code ~/} are returned as-is.
+     * </p>
+     * <h3>Examples:</h3>
+     * <p>
+     * Given {@code baseDir} equal to "/path/to/basedir" and a {@code volume} string equal to
+     * "./reldir:/some/other/dir", this method returns {@code /path/to/basedir/reldir:/some/other/dir}
+     * </p>
+     * <p>
+     * Given {@code baseDir} equal to "/path/to/basedir" and a {@code volume} string equal to
+     * "~/reldir:/some/other/dir", this method returns {@code /path/to/user/home/reldir:/some/other/dir}
+     * </p>
+     *
+     * @param baseDir the base directory used to resolve paths beginning with {@code ./}; <em>must</em> be absolute
+     * @param volume the volume string from the docker-compose file
+     * @return the volume string, with any relative paths resolved as absolute paths
+     */
+    static String resolveRelativeVolumeBinding(File baseDir, String volume) {
```

Comment:
It appears that relative paths in volume binding strings can only appear in `docker-compose.yaml` files. 

The Docker [Run reference](https://docs.docker.com/engine/reference/run/#volume-shared-filesystems) indicates that relative paths are _not_ allowed in image `VOLUME`s:

> -v, --volume=[host-src:]container-dest[:<options>]: Bind mount a volume.
The comma-delimited `options` are [rw|ro], [z|Z],
[[r]shared|[r]slave|[r]private], and [nocopy].
The 'host-src' is an absolute path or a name value.

And later in the same section:

> The container-dest must always be an absolute path such as /src/docs. The host-src can either be an absolute path or a name value. If you supply an absolute path for the host-dir, Docker bind-mounts to the path you specify. If you supply a name, Docker creates a named volume by that name.

> A name value must start with an alphanumeric character, followed by a-z0-9, _ (underscore), . (period) or - (hyphen). An absolute path starts with a / (forward slash).

> For example, you can specify either /foo or foo for a host-src value. If you supply the /foo value, Docker creates a bind-mount. If you supply the foo specification, Docker creates a named volume.

Let me know if you don't agree, but it seems that relative paths only need to be accounted for when using a `docker-compose.yml` file.

In that case, the base directory used to resolve relative paths should be the one defined in the compose handler:
```xml
  <external> 
     <type>compose</type> 
     <basedir>src/main/docker</basedir> 
     <composeFile>docker-compose.yml</composeFile>
  </external>
```

Overall, I'm not sure of the wider applicability this code.  What are your thoughts?  Maybe I am missing something.  I can certainly clean up the PR a bit, and have the resolution logic ready to be used by other classes as needed.